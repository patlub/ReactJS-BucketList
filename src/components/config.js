import axios from 'axios'


const axiosInstance = axios.create({
    baseURL: 'https://patrickluboobi-bucket-list-api.herokuapp.com',
});

export default axiosInstance
/**
 * Created by PATRICK on 8/17/2017.
 */
