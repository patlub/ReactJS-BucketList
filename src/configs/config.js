
/**
 * Created by PATRICK on 8/17/2017.
 */
export const baseURL = 'https://awesome-bucket.herokuapp.com';

