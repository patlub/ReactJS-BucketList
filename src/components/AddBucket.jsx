import React, {Component} from 'react';

// styles
import '../App.css';

// configs
import {baseURL} from './config';

// Third party
import axios from 'axios'
import {NotificationContainer, NotificationManager} from 'react-notifications';
import 'react-notifications/lib/notifications.css';

const initialState = {
    bucket: '',
    desc: '',
};

class AddBucket extends Component {
    constructor(props) {
        super(props);
        this.state = {
            loggedIn: localStorage.getItem('token'),
            bucket: '',
            desc: '',
        };
    }

    resetState = () => {
        this.setState(initialState);
    };

    /*
    * Fired when input changes
    * @param (event) event when input changes
    * */
    onInputChanged = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        })
    };

    /**
     * This method handles adding a bucket
     * @param {event} event when triggering addBucket action.
     */
    onAddBucketHandler = (event) => {
        event.preventDefault();

        axios.defaults.headers.common['Authorization'] = localStorage.getItem('token');
        axios.post(`${baseURL}/buckets`,
            {
                bucket: this.state.bucket,
                desc: this.state.desc
            })
            .then((response) => {
                this.props.addBucket(response.data);
                NotificationManager.success(`Bucket ${this.state.bucket} has been created`, `Success`);
                // Empty state
                this.resetState()
            })
            .catch((error) => {
                NotificationManager.error(`Bucket ${this.state.bucket} already exists`, `Error`);
            });
    };

    render() {
        return (
            <form className="form-inline" onSubmit={this.onAddBucketHandler}>
                <div className="form-group">
                    <input
                        className="form-control"
                        type="text"
                        name="bucket"
                        value={this.state.bucket}
                        onChange={this.onInputChanged}
                        placeholder="Bucket name"
                        required
                    />
                </div>
                <div className="form-group">
                    <input
                        className="form-control"
                        type="text"
                        name="desc"
                        value={this.state.desc}
                        onChange={this.onInputChanged}
                        placeholder="Description"
                        required
                    />
                </div>
                <div className="form-group">
                    <button type="submit" className="btn btn-primary">Add Bucket
                    </button>
                </div>
                <NotificationContainer/>
            </form>
        );
    }
}

export default AddBucket;
