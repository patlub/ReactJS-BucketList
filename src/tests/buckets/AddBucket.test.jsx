import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import AddBucket from '../../buckets/AddBucket';
import { baseURL } from '../../configs/config';

jest.mock('react-notifications');

const axios = require('axios');
const MockAdapter = require('axios-mock-adapter');

// This sets the mock adapter on the default instance
const mock = new MockAdapter(axios);
// Mock Login POST request to /login
mock.onPost(`${baseURL}/buckets`).reply(200, {
});

global.localStorage = {
  getItem: () => {},
  setItem: () => {},
};

describe('Component: Login', () => {
  const addBucket = (param) => {};
  const addBucketComponent = mount(<AddBucket addBucket={addBucket} />);
  const bucketNameInput = addBucketComponent.find('[type="text"]').at(0);
  const bucketDescInput = addBucketComponent.find('[type="text"]').at(1);
  const addBucketButton = addBucketComponent.find('[type="submit"]');

  it('Displays AddBucket component', () => {
    const rendered = renderer.create(
      <AddBucket />,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('updates state on input change', () => {
    bucketNameInput.simulate('change', { target: { name: 'name', value: 'Travel' } });
    bucketDescInput.simulate('change', { target: { name: 'desc', value: 'cities' } });
    expect(addBucketComponent.state().name).toEqual('Travel');
    expect(addBucketComponent.state().desc).toEqual('cities');
  });
  it('Adds a bucket', () => {
    addBucketButton.simulate('submit');
  });
});
