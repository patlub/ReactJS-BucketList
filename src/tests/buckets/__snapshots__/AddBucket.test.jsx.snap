// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Login Displays AddBucket component 1`] = `
<form
  className="form-inline"
  onSubmit={[Function]}
>
  <div
    className="form-group add-text-box"
  >
    <input
      className="form-control add-text-box"
      name="bucket"
      onChange={[Function]}
      placeholder="Bucket name"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group add-text-box"
  >
    <input
      className="form-control"
      name="desc"
      onChange={[Function]}
      placeholder="Description"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-primary"
      type="submit"
    >
      Add Bucket
    </button>
  </div>
</form>
`;
